### ISSUE TITLE
Decompiler sometimes outputs 'this' instead 'super'

### ISSUE THREAD

**Issue by ShadauxCat** [jac3km4/redscript#50]
For example, try adding the following code:

```swift
@replaceMethod(MeleeDeflectAttackEvents)
  public func OnUpdate(timeDelta: Float, stateContext: ref<StateContext>, scriptInterface: ref<StateGameScriptInterface>) -> Void {
    let slowMoStart: Float = this.GetStaticFloatParameterDefault("slowMoStart", 0.10);
    if !this.m_slowMoSet && this.GetInStateTime() > slowMoStart && !this.IsTimeDilationActive(stateContext, scriptInterface, n"") {
      scriptInterface.GetTimeSystem().SetTimeDilation(n"deflect", this.GetStaticFloatParameterDefault("slowMoAmount", 0.10), this.GetStaticFloatParameterDefault("slowDuration", 0.10), this.GetStaticCNameParameterDefault("slowMoEaseIn", n"Linear"), this.GetStaticCNameParameterDefault("slowMoEaseOut", n"Linear"));
      this.m_slowMoSet = true;
    };
    this.OnUpdate(timeDelta, stateContext, scriptInterface);
  }
```

This results in a crash when doing a perfect block/parry on a melee attack. However, this code was copied directly from the dump created by redscript-cli - it's theoretically what the game script is already doing.

It seems like the game script is calling the parent class implementation when it calls `this.OnUpdate()`, but doing the same thing from a redscript file seems to cause the function to call itself instead. Which... then causes it to call itself... and then call itself... and then call itself... and then call itself... until the game crashes, with a stack trace that reports a stack overflow.

**Comment by jac3km4** [comment]
you're right, this function seems to be calling itself, that's why:
`this.OnUpdate(timeDelta, stateContext, scriptInterface);`
needs to be replaced with
`super.OnUpdate(timeDelta, stateContext, scriptInterface);`

this looks like an issue with the decompiler, but if you run in any other similar case, you can just replace `this` with `super`.

**Comment by ShadauxCat** [comment]
Ok, cool. I wasn't sure if it was the decompiler or the compiler at fault - just knew that what the decompiler produced couldn't be run. :) I did think the decompiler output was weird because I didn't know how one was supposed to implement a recursive function if `this` automatically meant `super` with the same function name.

**Comment by ShadauxCat** [comment]
(And wasn't sure what the right way to call the parent was if the decompiler output was wrong. Thanks for answering that!)

**Comment by jac3km4** [comment]
thanks for the report, I'll change this issue to track a decompiler bug

**Comment by jac3km4** [comment]
should be fixed with eea9a2aa56a498d3f047ae8c56aca2124530cdde
